homeassistant/components/device_tracker/tplink.py

```diff
@@ -46,6 +60,45 @@ def get_scanner(hass, config):
 
 
 class TplinkDeviceScanner(DeviceScanner):
+    """Queries the router for connected devices."""
+
+    def __init__(self, config):
+        """Initialize the scanner."""
+        self.host = config[CONF_HOST]
+        self.password = config[CONF_PASSWORD]
+        self.username = config[CONF_USERNAME]
+
+        self.last_results = {}
+        self.success_init = self._update_info()
+
+    def scan_devices(self):
+        """Scan for new devices and return a list with found device IDs."""
+        self._update_info()
+        return self.last_results.keys()
+
+    def get_device_name(self, device):
+        """Get the name of the device."""
+        return self.last_results.get(device)
+
+    def _update_info(self):
+        """Ensure the information from the TP-Link router is up to date.
+
+        Return boolean if scanning successful.
+        """
+        from tplink.tplink import TpLinkClient
+        _LOGGER.info("Loading wireless clients...")
+        client = TpLinkClient(
```

Comment:
Do we need to create a new client on each update.

Reply:
My idea is that on home assistant the client should be just one. Current clients should be gradually migrated on the tplink package. I migrated the one I own and could test

Reply:
Yes, but on this one a new client instance is created each time the device tracker scans for new devices. That seems possibly unessecary to me. Would it work to create the client in init method, save the client as an instance attribute, and reuse the same client instance on each update?

Reply:
sure, seems a reasonable request.